lib/matplotlib/transforms.py

```diff
@@ -682,6 +682,77 @@ def intersection(bbox1, bbox2):
 class Bbox(BboxBase):
     """
     A mutable bounding box.
+
+    Examples
+    --------
+    **Create from known bounds**
+
+    The default constructor takes the boundary "points" ``[[xmin, ymin],
+    [xmax, ymax]]``.
+
+        >>> Bbox([[1, 1], [3, 7]])
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    Alternatively, a Bbox can be created from the flattened array, the
+    so-called "extents" ``(xmin, ymin, xmax, ymax)``
+
+        >>> Bbox.from_extents(1, 1, 3, 7)
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    or from the "bounds" ``(xmin, ymin, width, height)``.
+
+        >>> Bbox.from_bounds(1, 1, 2, 6)
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    **Create from collections of points**
+
+    The "empty" object for accumulating Bboxs is the null bbox, which has
+    ``(min, max) = (np.inf, -np.inf)`` for each dimension.
+
+        >>> Bbox.null()
+        Bbox([[inf, inf], [-inf, -inf]])
+
+    Adding points to the null bbox will give you the bbox of those points.
+
+        >>> box = Bbox.null()
+        >>> box.update_from_data_xy([[1, 1]])
+        >>> box
+        Bbox([[1.0, 1.0], [1.0, 1.0]])
+        >>> box.update_from_data_xy([[2, 3], [3, 2]], ignore=False)
+        >>> box
+        Bbox([[1.0, 1.0], [3.0, 3.0]])
+
+    Setting ``ignore=True`` is equivalent to starting over from a null bbox.
+
+        >>> box.update_from_data_xy([[1, 1]], ignore=True)
+        >>> box
+        Bbox([[1.0, 1.0], [1.0, 1.0]])
+
+    .. warning:: Not setting ``ignore`` can have unexpected results.
```

Comment:
Can you be more specific about this? Are you referring to the somewhat questionable behavior that the last state of `ignore` is used by default? If so, use something like:

> It is recommended to always specify `ignore` explicitly. If not set, the value of ignore from the previous call will be reused.

Reply:
I'm even more worried about users trying to *use* the fact that it's supposed to be the previous call's ignore value that is reused. 

In practice, `.ignore` could have been called, or external code that touched the bbox could have changed the default value. 

That happened to me when coding #16832, and it was a pain to track down. 

I updated the warning to be very specific.